Blockzilla/BrowserViewController.swift

```diff
@@ -976,7 +949,31 @@ extension BrowserViewController: URLBarDelegate {
     }
 
     func urlBarDidTapShield(_ urlBar: URLBar) {
-        showDrawer()
+        Telemetry.default.recordEvent(TelemetryEvent(category: TelemetryEventCategory.action, method: TelemetryEventMethod.open, object: TelemetryEventObject.trackingProtectionDrawer))
+        
+        switch trackingProtectionStatus {
+        case .on(let info):
```

Comment:
This feels like a lot of code duplication. Is there any way we could create a helper function that builds these for us so we can re-use it in both cases?